src/models.zig

```diff
@@ -153,6 +165,193 @@ pub const Model = struct {
 		return modelIndex;
 	}
 
+	pub fn loadModel(data: []const u8) u16 {
+		var vertices = main.List(Vec3f).init(main.stackAllocator);
+		defer vertices.deinit();
+
+		var normals = main.List(Vec3f).init(main.stackAllocator);
+		defer normals.deinit();
+
+		var uvs = main.List(Vec2f).init(main.stackAllocator);
+		defer uvs.deinit();
+
+		var tris = main.List(Triangle).init(main.stackAllocator);
+		defer tris.deinit();
+
+		var quadFaces = main.List(Quad).init(main.stackAllocator);
+		defer quadFaces.deinit();
+
+		var fixed_buffer = std.io.fixedBufferStream(data);
+		var buf_reader = std.io.bufferedReader(fixed_buffer.reader());
+		var in_stream = buf_reader.reader();
+		var buf: [128]u8 = undefined;
+		while (in_stream.readUntilDelimiterOrEof(&buf, '\n') catch |e| blk: {
+			std.log.err("Error reading line while loading model: {any}", .{e});
+			break :blk null;
+		}) |lineUntrimmed| {
+			if (lineUntrimmed.len < 3)
+				continue;
+			
+			var line = lineUntrimmed;
+			if (line[line.len - 1] == '\r') {
+				line = line[0..line.len - 1];
+			}
+			
+			if (line[0] == '#')
+				continue;
+			
+			if (std.mem.eql(u8, line[0..2], "v ")) {
+				var coordsIter = std.mem.split(u8, line[2..], " ");
+				var coords: Vec3f = undefined;
+				var i: usize = 0;
+				while (coordsIter.next()) |coord| : (i += 1) {
+					coords[i] = std.fmt.parseFloat(f32, coord) catch |e| blk: { std.log.err("Failed parsing {s} into float: {any}", .{coord, e}); break :blk 0; };
+				}
+				const coordsCorrect: Vec3f = .{coords[0] + 0.5, -coords[2] + 0.5, coords[1]};
+				vertices.append(coordsCorrect);
+			} else if (std.mem.eql(u8, line[0..3], "vn ")) {
+				var coordsIter = std.mem.split(u8, line[3..], " ");
+				var norm: Vec3f = undefined;
+				var i: usize = 0;
+				while (coordsIter.next()) |coord| : (i += 1) {
+					norm[i] = std.fmt.parseFloat(f32, coord) catch |e| blk: { std.log.err("Failed parsing {s} into float: {any}", .{coord, e}); break :blk 0; };
+				}
+				const normCorrect: Vec3f = .{norm[0], -norm[2], norm[1]};
+				normals.append(normCorrect);
+			} else if (std.mem.eql(u8, line[0..3], "vt ")) {
+				var coordsIter = std.mem.split(u8, line[3..], " ");
+				var uv: Vec2f = undefined;
+				var i: usize = 0;
+				while (coordsIter.next()) |coord| : (i += 1) {
+					uv[i] = std.fmt.parseFloat(f32, coord) catch |e| blk: { std.log.err("Failed parsing {s} into float: {any}", .{coord, e}); break :blk 0; };
+				}
+				uv[0] *= 4;
+				uv[1] *= 4;
+				uvs.append(.{uv[0], uv[1]});
+			} else if (std.mem.eql(u8, line[0..2], "f ")) {
+				const n = std.mem.count(u8, line[2..], " ") + 1;
+				switch (n) {
```

Comment:
I think it would be shorter and more readable to just do the iteration first, putting the vertices into a buffer of size 4, and then at the end check how many vertices you actually got.